sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/SupportsCustomSchemaWrite.java

```diff
@@ -27,12 +28,12 @@
  * @since 3.4.1
  */
 @Evolving
-public interface SupportsCustomSchemaWrite {
+public interface SupportsCustomSchemaWrite extends Table {
     /**
      * Represents a table with a custom schema to use for resolving DEFAULT column references when
      * inserting into the table. For example, this can be useful for excluding hidden pseudocolumns.
      *
      * @return the new schema to use for this process.
      */
-    StructType customSchemaForInserts();
+    Column[] customColumnsForInserts();
```

Comment:
This is a breaking change and we need to avoid it.

Reply:
oh it's not released yet

Reply:
Yes, it was added in https://github.com/apache/spark/pull/40996, for 3.4.1 and 3.5.0

Reply:
LGTM